crates/adapters/src/controller/sync.rs

```diff
@@ -115,6 +107,35 @@ fn pull_and_gc(
     }
 }
 
+pub fn pull_necessary(storage: &CircuitStorageConfig) -> Option<&SyncConfig> {
```

Comment:
Could you add a function description here: what is being pulled and why is it necessary?